docs/manual/src/Upgrading.md

```diff
@@ -61,6 +60,20 @@ Edge-cases broken include:
 
 See [Remote and External Types](./types/remote_ext_types.md) for more detail.
 
+## External Custom Types
+
+Previously you could describe an external Custom Type `Url` in UDL as:
+```
+[External="crate_name"]
+typedef extern Url;
+```
+
+But now you must use:
+```
+[Custom="crate_name"]
+typedef string Url; // replace `string` with any appropriate type.
+```
```

Comment:
I haven't looked over everything, but I think the main question is how this should be spelled out in the UDL.  The main alternative in my mind is something like:

```
[External="crate_name"]
typedef custom Url;
```

The upside is that this looks like other external type definitions, the downside is that it doesn't define the bridged type.  Maybe that's okay though.  I think we can get that info from the metadata when `--library` is used.

The other case is when `--library` isn't used.  In that case, maybe users can specify the bridged type with another attribute:

```
[External="crate_name", BridgeType="string"]
typedef custom Url;
```

This feels worse than the proposed syntax, but IMO bindgen without library mode is outdated and we should start deprecating it and/or making it somewhat of a second-class citizen.

Reply:
> The upside is that this looks like other external type definitions, the downside is that it doesn't define the bridged type. Maybe that's okay though. I think we can get that info from the metadata when --library is used.

I don't think this is true - I actually started thinking it was, and started making the builtin elt of Custom an Option<Type> - the problem is that this falls apart in the Rust scaffolding - it needs to know the type - even for external types - to know how to lower etc it.

>This feels worse than the proposed syntax, but IMO bindgen without library mode is outdated

Agree entirely - I really would prefer to have found a way to not specify the type, but failed (a few times actually :)

Reply:
> the problem is that this falls apart in the Rust scaffolding

oops, I should have clarified - we work around that in the rust scaffolding today by assuming a `RustBuffer`, but that falls apart in #2025.

Reply:
I tried for a while to make this work.  I made some things work, but I hit a fundamental roadblock: if the user is not using library mode, then we need to generate the bindings directly from the UDL.  In that case, there needs to be a way in the UDL to specify the builtin-type, since that's going to determine the FFI type.  This proposal seems as good as any other to me.